src/libAtoms/ScaLAPACK.f95

```diff
@@ -1461,4 +1474,122 @@ subroutine ScaLAPACK_to_array2d(A_info, A_data, array)
 #endif
 end subroutine ScaLAPACK_to_array2d
 
+! returns 64bit minimal work array length for pdgeqrf from 32bit sources
+! adapted from documentation of pdgeqrf
+function get_lwork_pdgeqrf_i32o64(m, n, ia, ja, mb_a, nb_a, &
+    myrow, mycol, rsrc_a, csrc_a, nprow, npcol) result(lwork)
+  integer, intent(in) :: m, n, ia, ja, mb_a, nb_a
+  integer, intent(in) :: myrow, mycol, rsrc_a, csrc_a, nprow, npcol
+  integer(idp) :: lwork
+
+  integer :: iarow, iacol, iroff, icoff
+  integer(idp) :: mp0, nq0, nb64
+
```

Comment:
Shouldn't lwork get a return value if scalapack is not set>?

Reply:
Good point. Thought I had done that. Thanks for checking. Fixed.